### ISSUE TITLE
Test suite fails run with Ruby 2.2

### ISSUE THREAD

**Issue by voxik** [freerange/mocha#214]
~~~
  1) Failure:
StubbingNilTest#test_should_allow_stubbing_method_on_nil [/builddir/build/BUILD/rubygem-mocha-1.1.0/usr/share/gems/gems/mocha-1.1.0/test/acceptance/stubbing_nil_test.rb:21]:
Test failed unexpectedly with message: [#<MinitestResult::Failure:0x0000000180c360 @failure=#<Minitest::UnexpectedError: can't stub method on frozen object: nil>>]


  2) Failure:
StubbingNilTest#test_should_prevent_stubbing_method_on_nil [/builddir/build/BUILD/rubygem-mocha-1.1.0/usr/share/gems/gems/mocha-1.1.0/test/acceptance/stubbing_nil_test.rb:40]:
Failed assertion, no message given.


  3) Failure:
StubbingNilTest#test_should_warn_on_stubbing_method_on_nil [/builddir/build/BUILD/rubygem-mocha-1.1.0/usr/share/gems/gems/mocha-1.1.0/test/acceptance/stubbing_nil_test.rb:30]:
Test failed unexpectedly with message: [#<MinitestResult::Failure:0x000000017ecc40 @failure=#<Minitest::UnexpectedError: can't stub method on frozen object: nil>>]


  4) Failure:
StubbingNilTest#test_should_default_to_prevent_stubbing_method_on_non_mock_object [/builddir/build/BUILD/rubygem-mocha-1.1.0/usr/share/gems/gems/mocha-1.1.0/test/acceptance/stubbing_nil_test.rb:48]:
Failed assertion, no message given.
~~~

nil is frozen in Ruby 2.2 so Mocha behavior should be probably adjusted.

**Comment by floehopper** [comment]
@voxik: Thanks for reporting this. I'll try to fix it as soon as I have time.

**Comment by floehopper** [comment]
The specific test failures mentioned above were fixed in a65ea1ed3dce43cbc2cc16b3660afd2cce3db33e.

You can see the tests passing with Ruby v2.2 on Travis CI e.g. in [this build](https://travis-ci.org/freerange/mocha/jobs/52801355).

Closing.
